### PR TITLE
Support API v4 for user management

### PR DESCRIPTION
**Pull request by gaetan-craft**
User management changed a bit with v4 of graylog-server. This PRs fixes the creation and update of user in new version while keeping old version.
I'm not a seasoned go developer, feel free to suggest enhancement

### CODE REVIEW

**Comment by zahiar** [comment]
@gaetan-craft Thanks for this. I'll take a look over the weekend 🙂

File: graylog/resource/user/create.go [hunk 1]
```diff
@@ -15,16 +15,30 @@ func create(d *schema.ResourceData, m interface{}) error {
...
 	}
 
-	if _, err := cl.User.Create(ctx, data); err != nil {
+	_, err = cl.User.Create(ctx, data)
+	if err != nil {
 		return err
 	}
-	d.SetId(data[keyUsername].(string))
+	if cl.APIVersion == "v4" {
+		new_data, _, _ := cl.User.Get(ctx, data[keyUsername].(string), "v3")
```
Comment (zahiar) [inline graylog/resource/user/create.go:33]: Should the last parameter `v3` be `v4`, as we are within the `v4` IF block?

File: graylog/resource/user/resource.go [hunk 2]
```diff
@@ -36,8 +36,23 @@ func Resource() *schema.Resource {
...
 			"full_name": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:          schema.TypeString,
+				Optional:      true,
+				Required:      false,
+				Computed:      true,
+				ConflictsWith: []string{"first_name", "last_name"},
+			},
+			"first_name": {
```
Comment (zahiar) [inline graylog/resource/user/resource.go:45]: Could you updated the docs for this resource to add the new `first_name` & `last_name`, as well as mentioning they conflict with `full_name`?
Docs file can be found here: `docs/resources/user.md`

**Review by zahiar — CHANGES_REQUESTED** [review]

**Comment by zahiar** [comment]
@gaetan-craft Thanks for this - couple of items to address & then it'll be good to merge/release 🎉

File: graylog/resource/user/create.go (hunk 1 above)
Comment (gaetan-craft) [reply graylog/resource/user/create.go:34]: It's due to the strangeness of the graylog api which does not send back the id on creation

File: graylog/resource/user/resource.go (hunk 2 above)
Comment (gaetan-craft) [reply graylog/resource/user/resource.go:45]: Sure, done

File: graylog/resource/user/create.go (hunk 1 above)
Comment (zahiar) [reply graylog/resource/user/create.go:34]: 👍

File: graylog/resource/user/resource.go (hunk 2 above)
Comment (zahiar) [reply graylog/resource/user/resource.go:45]: 👍

**Comment by zahiar** [comment]
@gaetan-craft looks like some of the acceptance tests are failing, you can try and fix them or if you prefer, I can look over them when I get a chance.

**Comment by gaetan-craft** [comment]
> @gaetan-craft looks like some of the acceptance tests are failing, you can try and fix them or if you prefer, I can look over them when I get a chance.
I'll give it a try in the next days

**Comment by gaetan-craft** [comment]
@zahiar done, thanks for the review

**Pull request merged**